src/catchup/CatchupTransactionsWork.cpp

```diff
@@ -69,22 +63,14 @@ CatchupTransactionsWork::onSuccess()
         return WORK_PENDING;
     }
 
-    // Phase 2: download and decompress the transactions.
-    if (!mDownloadTransactionsWork)
+    // Phase 2: download and apply the transactions.
+    if (!mDownloadAndApplyTransactionsWork)
     {
-        CLOG(INFO, "History") << "Catchup " << mCatchupTypeName
-                              << " downloading transactions";
-        mDownloadTransactionsWork = addWork<BatchDownloadWork>(
-            mRange, HISTORY_FILE_TYPE_TRANSACTIONS, mDownloadDir);
-        return WORK_PENDING;
-    }
-
-    // Phase 3: apply the transactions.
-    if (!mApplyWork)
-    {
-        CLOG(INFO, "History") << "Catchup " << mCatchupTypeName
-                              << " applying history";
-        mApplyWork = addWork<ApplyLedgerChainWork>(mDownloadDir, mRange);
+        CLOG(INFO, "History")
+            << "Catchup downloading and applying transactions for range ["
+            << mRange.first() << ".." << mRange.last() << "]";
+        mDownloadAndApplyTransactionsWork =
+            addWork<DownloadAndApplyTransactionsWork>(mRange, mDownloadDir);
```

Comment:
similar comment than earlier, this particular refactor doesn't really achieve anything as it introduces a new work type for no particular reason (only used in one place)